system/Cookie/BaseCookie.php

```diff
@@ -0,0 +1,306 @@
+		{
+			throw CookieException::forInvalidSameSite($this->samesite);
+		}
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Set cookie prefix.
+	 *
+	 * @param string $prefix
+	 *
+	 * @return $this
+	 */
+	public function setPrefix(string $prefix)
+	{
+		$this->prefix = $prefix;
+
+		return $this;
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Get cookie prefix.
+	 *
+	 * @return string
+	 */
+	public function getPrefix(): string
+	{
+		return $this->prefix;
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Set cookie path.
+	 *
+	 * @param string $path
+	 *
+	 * @return $this
+	 */
+	public function setPath(string $path)
+	{
+		$this->path = $path;
+
+		return $this;
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Get cookie path.
+	 *
+	 * @return string
+	 */
+	public function getPath(): string
+	{
+		return $this->path;
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Set cookie domain.
+	 *
+	 * @param string $domain
+	 *
+	 * @return $this
+	 */
+	public function setDomain(string $domain)
+	{
+		$this->domain = $domain;
+
+		return $this;
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Get cookie domain.
+	 *
+	 * @return string
+	 */
+	public function getDomain(): string
+	{
+		return $this->domain;
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Set cookie secure.
+	 *
+	 * @param boolean $secure
+	 *
+	 * @return $this
+	 */
+	public function setSecure(bool $secure = true)
+	{
+		$this->secure = $secure;
+
+		return $this;
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Check cookie secure status.
+	 *
+	 * @return boolean
+	 */
+	public function isSecure(): bool
+	{
+		return $this->secure;
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Set cookie httponly.
+	 *
+	 * @param boolean $httponly
+	 *
+	 * @return $this
+	 */
+	public function setHTTPOnly(bool $httponly = true)
+	{
+		$this->httponly = $httponly;
+
+		return $this;
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Check cookie httponly status.
+	 *
+	 * @return boolean
+	 */
+	public function isHTTPOnly(): bool
+	{
+		return $this->httponly;
+	}
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Set cookie samesite.
+	 * 
+	 * Set value to the default configuration, if $samesite is invalid.
+	 * 
+	 * @param string $samesite
+	 *
+	 * @return $this
+	 */
+	public function setSameSite(string $samesite)
+	{
+		if (! in_array(strtolower($samesite), ['none', 'lax', 'strict', ''], true))
+		{
+			$samesite = $this->properties['samesite'];
+		}
+
+		$this->samesite = ucfirst(strtolower($samesite));
```

Comment:
I think we should be consistent in throwing exceptions here if wrong samesite is given. That way, developers won't get confused why there are setting this value but another one gets assigned. It's like telling them directly the problem they're doing.